inference-engine/ie_bridges/python/tests/test_IECore.py

```diff
@@ -253,3 +255,22 @@ def test_net_from_buffer_valid():
     o_net2 = ref_net.outputs
     assert ii_net.keys() == ii_net2.keys()
     assert o_net.keys() == o_net2.keys()
+
+
+@pytest.mark.skipif(os.environ.get("TEST_DEVICE","CPU") != "GPU", reason=f"Device dependent test")
+def test_load_network_release_gil(device):
+    def detect_long_gil_holds():
+        sleep_time = 0.01
+        latency_alert_threshold = 0.1
+        while True:
+            start_sleep = time()
+            sleep(sleep_time)
+            elapsed = time() - start_sleep
+            assert elapsed < latency_alert_threshold
+
+    ie = IECore()
+    net = ie.read_network(model=test_net_xml, weights=test_net_bin)
+    gil_hold_detection_thread = Thread(daemon=True, target=detect_long_gil_holds)
+    gil_hold_detection_thread.start()
+    sleep(0.1)
+    exec_net = ie.load_network(net, device)
```

Comment:
Hi! I'm not familiar with OpenVINO's testing infrastructure, but I believe this test has two bugs:

1) It won't run the assert in the main thread, so `pytest` will never find out that a thread crashed (even though it will print in the logs)
2) The test will leave a stray thread running

Here's my suggested fix:
```suggestion
@pytest.mark.skipif(os.environ.get("TEST_DEVICE","CPU") != "GPU", reason=f"Device dependent test")
def test_load_network_release_gil(device):
    running = True
    message_queue = Queue()

    def detect_long_gil_holds():
        sleep_time = 0.01
        latency_alert_threshold = 0.1

        # Send a message to indicate the thread is running and ready to detect GIL locks
        message_queue.put("ready to detect")

        while running:
            start_sleep = time()
            sleep(sleep_time)
            elapsed = time() - start_sleep
            if elapsed > latency_alert_threshold:
                # Send a message to the testing thread that a long GIL lock occurred
                message_queue.put(latency_alert_threshold)

    ie = IECore()
    net = ie.read_network(model=test_net_xml, weights=test_net_bin)

    # Wait for the GIL lock detector to be up and running
    gil_hold_detection_thread = Thread(daemon=True, target=detect_long_gil_holds)
    gil_hold_detection_thread.start()
    assert message_queue.get(timeout=5) == "ready to detect"

    # Run the function that should unlock the GIL
    exec_net = ie.load_network(net, device)

    # Ensure resources are closed
    running = False
    gil_hold_detection_thread.join(timeout=5)

    # Assert there were never any long gil locks
    assert message_queue.qsize() == 0, \
        f"More than 0 GIL locks occured! Latency: {message_queue.get()})"
```

Reply:
Hi, you are right. I updated this test. Thank you for your comment!

> Hi! I'm not familiar with OpenVINO's testing infrastructure, but I believe this test has two bugs:
> 
> 1. It won't run the assert in the main thread, so `pytest` will never find out that a thread crashed (even though it will print in the logs)
> 2. The test will leave a stray thread running
> 
> Here's my suggested fix: